### ISSUE TITLE
should Observation.add be on the map or the computedProp

### ISSUE THREAD

**Issue by justinbmeyer** [canjs/can-define#149]
Related: https://github.com/canjs/can-map/issues/30

A consideration is that it might be harder to "trace" the dependency.

**Comment by justinbmeyer** [comment]
we should ignore reading the compute, and instead observe directly on the property.  This is likely a bit slower, but much more accurate.

**Comment by andrejewski** [comment]
@justinbmeyer I wrote some test cases based on https://github.com/canjs/can-map/issues/30 which all passed for `can-define`. Are you sure this is a problem for DefineMap and not just Map? If it is, I'd like some clarification on the problem here.

**Comment by justinbmeyer** [comment]
Assigning this to @bmomberger-bitovi .  @bmomberger-bitovi please close this out while you are working on can-define.  The call to `Observation.add` should be on the map, no the compute (or now observation).  This is so we know the map is bound to (things like can-connect's instance store depend on this).

**Comment by justinbmeyer** [comment]
```js
DefineMap.extend({
  prop: "string"
  get proped(){  return this.prop }
});

compute(function(){
  return map.proped // should call map.on("proped"), instead -> map._computed.proped.on("change")
})

```

**Comment by andrejewski** [comment]
@justinbmeyer I made the following test case:
```js
QUnit.test('should observe the map key, not value', function (assert) {
	var MyMap = DefineMap.extend({
		foo: 'string',
		get bar () {
			return this.foo;
		}
	});

	var map = new MyMap();
	var newValue = 'baz';
	map.foo = newValue;

	assert.equal(map.bar, newValue, 'map value updated');
});
```

And another using a compute, for some more indirection I suppose:

```js
QUnit.test('should observe the map key, not value', function (assert) {
	var MyMap = DefineMap.extend({
		foo: 'string',
		get bar () {
			return this.foo;
		}
	});

	var map = new MyMap();
	var newValue = 'baz';
	map.foo = newValue;

	var listeningCompute = compute(function () {
		return map.bar;
	});

	var value = listeningCompute();
	assert.equal(value, newValue, 'map value updated');
});
```

From what I can tell, the [`Observation.add` call is binding to the key](https://github.com/canjs/can-define/blob/master/map/map.js#L48) as desired. Perhaps @bmomberger-bitovi did fix this during the reflecting.

**Comment by phillipskevin** [comment]
@Sinjhin is going to roll back the changes for this from https://github.com/canjs/can-define/pull/244 because it is causing other problems: https://travis-ci.org/canjs/canjs/builds/268510138#L4083-L4098. We will need to to revisit this once we have a way of handling the asynchronous computes created by things like https://github.com/canjs/can-connect/blob/master/can/ref/ref.js#L300.

**Comment by phillipskevin** [comment]
https://github.com/canjs/can-observation/pull/109 was also rolled back for the same reason. This will need to be re-done after we have async computes.

**Comment by justinbmeyer** [comment]
Closed in 4.0: https://github.com/canjs/can-define/commit/15487345f850b2122bf189ea49801462b2f06171
